compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeWindowPanel.desktop.kt

```diff
@@ -114,13 +117,28 @@ internal class ComposeWindowPanel(
         composeContainer.preferredSize
     }
 
+    private val disableInputMethodsOnFocusGainListener: FocusListener = object : FocusAdapter() {
+        override fun focusGained(e: FocusEvent) {
+            e.component?.enableInputMethods(false)
```

Comment:
There can be a race between this and `fun enableInput(inputMethodRequests: InputMethodRequests)`, that can be called on recomposition that happens after `addNotify`, before `isVisible` (and probably before focusGained).

I think you need to store the current state in `DesktopPlatformComponent` and set it here.

Reply:
Right.

I now fixed it by adding a flag in `WindowSkiaLayerComponent` and `SwingSkiaLayerComponent`. This also fixes it for `SwingSkiaLayerComponent` (but again, only for JBR).